### PR TITLE
Fix random stats

### PR DESCRIPTION
**Pull request by rowleya**
Fixes some issues with random stats in PyNN 8

### CODE REVIEW

File: spynnaker8/models/connectors/multapse_connector.py [hunk 1]
```diff
@@ -22,23 +23,29 @@ class MultapseConnector(_BaseClass):
...
+    def __init__(self, n, allow_self_connections=True,
+                 with_replacement=True, safe=True, verbose=False,
+                 rng=None):
         super(MultapseConnector, self).__init__(
-            num_synapses=num_synapses,
-            allow_self_connections=allow_self_connections,
-            with_replacement=with_replacement, safe=safe, verbose=verbose)
+            num_synapses=n, allow_self_connections=allow_self_connections,
+            with_replacement=with_replacement, safe=safe, verbose=verbose,
+            rng=rng)
```
Comment (andrewgait) [inline spynnaker8/models/connectors/multapse_connector.py]: The base class doesn't have rng= in the arguments, so this falls over at the moment - I guess it should just be added to the list in the base class in sPyNNaker and then probably set self._rng in there too?

File: spynnaker8/models/connectors/multapse_connector.py [hunk 2]
```diff
@@ -22,23 +23,29 @@ class MultapseConnector(_BaseClass):
...
+        for j in range(0, size - 1):
+            multinomial[j] = self._rng.next(
+                1, distribution="binomial",
+                parameters={'n': dn, 'p': prob_connect[j] / total})
+            dn = dn - multinomial[j]
+            if dn <= 0:
+                break
+            total = total - prob_connect[j]
+        if dn > 0:
+            multinomial[size - 1] = dn
```
Comment (andrewgait) [inline spynnaker8/models/connectors/multapse_connector.py]: Ah, yes, I see - I wasn't that far off.  Knew there was probably a better way of doing it though!

File: spynnaker8/models/connectors/multapse_connector.py (hunk 1 above)
Comment (rowleya) [reply spynnaker8/models/connectors/multapse_connector.py]: Oops - this was in my sPyNNaker changes, but not here - I will make a matching branch...

Comment (rowleya) [reply spynnaker8/models/connectors/multapse_connector.py]: Added support in SpiNNakerManchester/sPyNNaker#528

File: spynnaker8/models/connectors/multapse_connector.py (hunk 2 above)
Comment (rowleya) [reply spynnaker8/models/connectors/multapse_connector.py]: I basically copied this from numpy!

**Pull request merged**